payments/providers/bambora_payform.py

```diff
@@ -204,24 +203,35 @@ def payload_add_products(self, payload: dict, order: Order, language: str):
 
     def payload_add_customer(self, payload: dict, order: Order):
         """Attach customer data to payload"""
-        if not order.lease or order.lease and not order.lease.application:
-            raise MissingCustomerError(_("Order is not associated with a Lease"))
-
-        application = order.lease.application
-
-        payload.update(
-            {
-                "email": application.email.strip(),
-                "customer": {
-                    "firstname": application.first_name.capitalize(),
-                    "lastname": application.last_name.capitalize(),
+        if hasattr(order, "lease") and order.lease.application:
+            application = order.lease.application
+            payload.update(
+                {
                     "email": application.email.strip(),
-                    "address_street": application.address,
-                    "address_zip": application.zip_code,
-                    "address_city": application.municipality.capitalize(),
-                },
-            }
-        )
+                    "customer": {
+                        "firstname": application.first_name.capitalize(),
+                        "lastname": application.last_name.capitalize(),
+                        "email": application.email.strip(),
+                        "address_street": application.address,
+                        "address_zip": application.zip_code,
+                        "address_city": application.municipality.capitalize(),
+                    },
+                }
+            )
```

Comment:
Should we update the fields on the `order` so that the information matches?

Reply:
Agreed that keeping it like it is.